### ISSUE TITLE
[jvm-packages] Building package on macos 

### ISSUE THREAD

**Issue by bourbaki** [dmlc/xgboost#4292]
Hello.
I tried to build JVM packages on macos:
```
export JAVA_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_192.jdk/Contents/Home
cd jvm-packages
mvn package
```
Then I tried to launch a spark example with the iris dataset:
```
spark-submit --jars xgboost4j-spark/target/xgboost4j-spark-0.83-SNAPSHOT.jar,xgboost4j/target/xgboost4j-0.83-SNAPSHOT.jar   --class ml.dmlc.xgboost4j.scala.example.spark.SparkTraining xgboost4j-example/target/xgboost4j-example-0.83-SNAPSHOT.jar ../../iris.data.txt
```
And I got the following error:
```
2019-03-24 19:18:37 WARN  TaskSetManager:66 - Lost task 1.0 in stage 5.0 (TID 6, localhost, executor driver): java.lang.UnsatisfiedLinkError: /private/var/folders/p2/v_hdhwds1m9gpgtdg0c_yhlwrbd_1p/T/libxgboost4j313286062428554987.dylib: dlopen(/private/var/folders/p2/v_hdhwds1m9gpgtdg0c_yhlwrbd_1p/T/libxgboost4j313286062428554987.dylib, 1): no suitable image found.  Did find:
	/private/var/folders/p2/v_hdhwds1m9gpgtdg0c_yhlwrbd_1p/T/libxgboost4j313286062428554987.dylib: unknown file type, first eight bytes: 0xEF 0xBF 0xBD 0xEF 0xBF 0xBD 0xEF 0xBF
	/private/var/folders/p2/v_hdhwds1m9gpgtdg0c_yhlwrbd_1p/T/libxgboost4j313286062428554987.dylib: unknown file type, first eight bytes: 0xEF 0xBF 0xBD 0xEF 0xBF 0xBD 0xEF 0xBF
	at java.lang.ClassLoader$NativeLibrary.load(Native Method)
	at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1941)
	at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1824)
	at java.lang.Runtime.load0(Runtime.java:809)
	at java.lang.System.load(System.java:1086)
        ....
```
So there's something wrong with a packaged `dylib` file. After further investigation, I found a strange thing. The`dylib` file in the jar is not the same as the file generated by `create_jni` into `resources/lib` directory. A quick look at the headers:
```
❯ hexdump -n 8  xgboost4j/src/main/resources/lib/libxgboost4j.dylib
0000000 cf fa ed fe 07 00 00 01
❯ hexdump -n 8  xgboost4j/target/classes/lib/libxgboost4j.dylib
0000000 ef bf bd ef bf bd ef bf
```
Looks like there's something that changes the header of the binary 😄(the rest of the contents are the same). Do you have any idea what could be the cause of it?

```
Java: java version "1.8.0_192"
Maven: Apache Maven 3.6.0
XGBoost: 5aa42b5f
macos:10.13.16
```

**Comment by hcho3** [comment]
@bourbaki I think #4303 is related.

**Comment by hcho3** [comment]
Addressed in #4303.
